diesel/src/connection/cursor.rs

```diff
@@ -6,15 +6,17 @@ use std::marker::PhantomData;
 
 /// The type returned by various [`Connection`](struct.Connection.html) methods.
 /// Acts as an iterator over `T`.
-pub struct Cursor<ST, T> {
+pub struct Cursor<ST, T, R> {
     current_row: usize,
-    db_result: DbResult,
+    db_result: R,
     _marker: PhantomData<(ST, T)>,
 }
 
-impl<ST, T> Cursor<ST, T> {
+impl<ST, T, R> Cursor<ST, T, R> where
```

Comment:
While we may go this route in the future, to limit the scope of this PR, I'd prefer that we did not change this type, and instead had the various methods on `Connection` return `Box<Iterator>`